sktime/forecasting/hf_patchtst_forecaster.py

```diff
@@ -0,0 +1,548 @@
+
+if _check_soft_dependencies("torch", severity="none"):
+    import torch
+    from torch.utils.data import Dataset
+else:
+
+    class Dataset:
+        """Dummy class if torch is unavailable."""
+
+        pass
+
+
+if _check_soft_dependencies("transformers", severity="none"):
+    from transformers import (
+        PatchTSTConfig,
+        PatchTSTForPrediction,
+        PatchTSTModel,
+        Trainer,
+        TrainingArguments,
+    )
+
+
+class HFPatchTSTForecaster(_BaseGlobalForecaster):
+    """Interface for the PatchTST forecaster.
+
+    This model has 3 available modes:
+        1) Loading an full PatchTST model and pretrain on some dataset.
+        This can be done by passing in a PatchTST config dictionary or passing
+        in available parameters in this estimator, or by loading the
+        HFPatchTSTForecaster without any passed arguments.
+
+        2) Load a pre-trained model for fine-tuning. The parameter `model_path`
+        can be used to load a pre-trained model for fine-tuning. Using a
+        pretrained model will override any other model config parameters, and
+        you can use the `fit` function to then fine-tune your pretrained model.
+        See the `model_path` docstring for more details.
+
+        3) Load a pre-trained model for zero-shot forecasting. The parameter
+        `model_path` can be used to load a pre-trained model for fine-tuning.
+        Using a pretrained model will override any other model config
+        parameters, and you can use the `predict` function to do zero-shot
+        forecasting. See the `model_path` docstring for more details.
+
+    Parameters
+    ----------
+    model_path : str, optional
+        Path to the Huggingface model to use for global forecasting. If
+        model_path is passed, the remaining model config parameters will be
+        ignored except for specific training or dataset parameters.
+        This has 3 options:
+            - model id to an online pretrained PatchTST Model hosted on HuggingFace
+            - A path or url to a saved configuration JSON file
+            - A path to a *directory* containing a configuration file saved
+            using the `~PretrainedConfig.save_pretrained` method
+            or the `~PreTrainedModel.save_pretrained` method
+    fit_strategy : str, values = ["full","minimal","zero-shot"], default = "full"
+        String to set the fit_strategy of the model.
+        If set to `full`, it will
+        re-initialize an full model with the specified config or
+        estimator arguments.
+        If set to "minimal" will use the `model_path`
+        argument and the passed in `y` in fit to fine-tune the model.
+        If set to "zero-shot", it will load the model in zero-shot forecasting
+        model with the argument `model_path` and ignore any passed `y`.
+        Note that both "minimal" and "zero-shot" fit_strategy requires a mandatory
+        passed in `model_path`.
```

Comment:
- "full" and "zero-shot" looks good, can you inform that "minimal" means to train only the changed weights
- also create bullets, to make it more readable